ui/packages/catalog-ui-search/src/main/webapp/react-component/container/time-settings/index.tsx

```diff
@@ -0,0 +1,219 @@
+    value: Common.getTimeZones()['-10'],
+  },
+  {
+    label: '-09:00',
+    value: Common.getTimeZones()['-9'],
+  },
+  {
+    label: '-08:00',
+    value: Common.getTimeZones()['-8'],
+  },
+  {
+    label: '-07:00',
+    value: Common.getTimeZones()['-7'],
+  },
+  {
+    label: '-06:00',
+    value: Common.getTimeZones()['-6'],
+  },
+  {
+    label: '-05:00',
+    value: Common.getTimeZones()['-5'],
+  },
+  {
+    label: '-04:00',
+    value: Common.getTimeZones()['-4'],
+  },
+  {
+    label: '-03:00',
+    value: Common.getTimeZones()['-3'],
+  },
+  {
+    label: '-02:00',
+    value: Common.getTimeZones()['-2'],
+  },
+  {
+    label: '-01:00',
+    value: Common.getTimeZones()['-1'],
+  },
+  {
+    label: 'UTC, +00:00',
+    value: Common.getTimeZones()['UTC'],
+  },
+  {
+    label: '+01:00',
+    value: Common.getTimeZones()['1'],
+  },
+  {
+    label: '+02:00',
+    value: Common.getTimeZones()['2'],
+  },
+  {
+    label: '+03:00',
+    value: Common.getTimeZones()['3'],
+  },
+  {
+    label: '+04:00',
+    value: Common.getTimeZones()['4'],
+  },
+  {
+    label: '+05:00',
+    value: Common.getTimeZones()['5'],
+  },
+  {
+    label: '+06:00',
+    value: Common.getTimeZones()['6'],
+  },
+  {
+    label: '+07:00',
+    value: Common.getTimeZones()['7'],
+  },
+  {
+    label: '+08:00',
+    value: Common.getTimeZones()['8'],
+  },
+  {
+    label: '+09:00',
+    value: Common.getTimeZones()['9'],
+  },
+  {
+    label: '+10:00',
+    value: Common.getTimeZones()['10'],
+  },
+  {
+    label: '+11:00',
+    value: Common.getTimeZones()['11'],
+  },
+  {
+    label: '+12:00',
+    value: Common.getTimeZones()['12'],
+  },
+]
+
+const TimeFormat = [
+  {
+    label: 'ISO 8601',
+    value: Common.getDateTimeFormats()['ISO'],
+  },
+  {
+    label: '24 Hour Standard',
+    value: Common.getDateTimeFormats()['24'],
+  },
+  {
+    label: '12 Hour Standard',
+    value: Common.getDateTimeFormats()['12'],
+  },
+]
+
+type UserPreferences = {
+  get: (key: string) => any
+  set: ({}) => void
+  savePreferences: () => void
+}
+
+const getUserPreferences = (): UserPreferences =>
+  user.get('user').get('preferences')
+
+const timeZoneModel = new Property({
+  label: 'Time Zone',
+  value: [getUserPreferences().get('timeZone')],
+  enum: TimeZones,
+})
+
+const timeFormatModel = new Property({
+  label: 'Time Format',
+  value: [getUserPreferences().get('dateTimeFormat')],
+  enum: TimeFormat,
+})
+
+const savePreferences = (model: {}) => {
+  const preferences = getUserPreferences()
+
+  preferences.set(model)
+
+  preferences.savePreferences()
+}
+
+const getCurrentDateTimeFormat = () =>
+  getUserPreferences().get('dateTimeFormat').datetimefmt
+const getCurrentTimeZone = () => getUserPreferences().get('timeZone')
+
+const getCurrentTime = (
+  format: string = getCurrentDateTimeFormat(),
+  timeZone: string = getCurrentTimeZone()
+) => momentTimezone.tz(moment(), timeZone).format(format)
+
+export default withListenTo(
```

Comment:
In order to prevent large diffs with indentation when we move to redux, please add your export default at the bottom of the file.